br/pkg/lightning/backend/local/key_adapter.go

```diff
@@ -46,36 +47,41 @@ func reallocBytes(b []byte, n int) []byte {
 
 type noopKeyAdapter struct{}
 
-func (noopKeyAdapter) Encode(dst []byte, key []byte, _ int64) []byte {
+func (noopKeyAdapter) Encode(dst []byte, key []byte, _ []byte) []byte {
 	return append(dst, key...)
 }
 
 func (noopKeyAdapter) Decode(dst []byte, data []byte) ([]byte, error) {
 	return append(dst, data...), nil
 }
 
-func (noopKeyAdapter) EncodedLen(key []byte) int {
+func (noopKeyAdapter) EncodedLen(key []byte, _ []byte) int {
 	return len(key)
 }
 
 var _ KeyAdapter = noopKeyAdapter{}
 
 type dupDetectKeyAdapter struct{}
 
-func (dupDetectKeyAdapter) Encode(dst []byte, key []byte, rowID int64) []byte {
+func (dupDetectKeyAdapter) Encode(dst []byte, key []byte, rowID []byte) []byte {
 	dst = codec.EncodeBytes(dst, key)
```

Comment:
I wonder if we can skip encoding the key first, since in all cases, the key is already memcomparable.

Reply:
Updated

Reply:
Did you forget to remove `dst = codec.EncodeBytes(dst, key)`?

Reply:
No, I reverted this change because a lot of tests failed and @gozssky is not quite sure about if this can be removed.

Reply:
Seems we can introduce `type EncodedKey []byte` in future, then we will have more confidence to know if it's encoded already